Tests/Microsoft.AppCenter.Crashes.Test.Windows/CrashesTest.cs

```diff
@@ -201,5 +213,164 @@ public void ProcessPendingErrorsDoesNotCrashOnFileDeletionFailure()
+        {
+            ErrorLogHelper.Instance = GenerateErrorLogHelperWithPendingFile();
+
+            // Start Crashes.
+            Crashes.SetEnabledAsync(true).Wait();
+            Crashes.ShouldAwaitUserConfirmation = () => { return false; };
+            Crashes.Instance.OnChannelGroupReady(_mockChannelGroup.Object, string.Empty);
+            Crashes.Instance.ProcessPendingErrorsTask.Wait();
+
+            // Verify the logs have not been processed and have been removed.
+            _mockChannel.Verify(channel => channel.EnqueueAsync(It.Is<ManagedErrorLog>(log => log.Id == expectedManagedErrorLog.Id && log.ProcessId == expectedProcessId)), Times.Once());
+            var alwaysSendValue = AppCenter.Instance.ApplicationSettings.GetValue(Crashes.PrefKeyAlwaysSend, true);
+            Assert.IsFalse(alwaysSendValue);
+            Mock.Get(ErrorLogHelper.Instance).Verify(instance => instance.InstanceRemoveStoredErrorLogFile(expectedManagedErrorLog.Id), Times.Once());
+        }
+
+        [TestMethod]
+        public void SendCrashReportsOrAwaitUserConfirmationAsyncTrueCallback()
+        {
+            ErrorLogHelper.Instance = GenerateErrorLogHelperWithPendingFile();
+
+            // Start Crashes.
+            Crashes.SetEnabledAsync(true).Wait();
+            Crashes.ShouldAwaitUserConfirmation = () => { return true; };
+            Crashes.Instance.OnChannelGroupReady(_mockChannelGroup.Object, string.Empty);
+            Crashes.Instance.ProcessPendingErrorsTask.Wait();
+
+            // Verify nothing in HandleUserConfirmationAsync has been called.
+            _mockChannel.Verify(channel => channel.EnqueueAsync(It.Is<ManagedErrorLog>(log => log.Id == expectedManagedErrorLog.Id && log.ProcessId == expectedProcessId)), Times.Never());
+            var alwaysSendValue = AppCenter.Instance.ApplicationSettings.GetValue(Crashes.PrefKeyAlwaysSend, true);
+            Assert.IsFalse(alwaysSendValue);
+            Mock.Get(ErrorLogHelper.Instance).Verify(instance => instance.InstanceRemoveStoredErrorLogFile(expectedManagedErrorLog.Id), Times.Never());
+        }
+
+        [TestMethod]
+        public void HandleUserConfirmationAsyncDontSend()
+        {
+            ErrorLogHelper.Instance = GenerateErrorLogHelperWithPendingFile();
+
+            // Start Crashes.
+            Crashes.SetEnabledAsync(true).Wait();
+            Crashes.ShouldAwaitUserConfirmation = () => { return true; }; // Avoids processing the logs during the ProcessPendingErrorsTask
+            Crashes.Instance.OnChannelGroupReady(_mockChannelGroup.Object, string.Empty);
+            Crashes.Instance.ProcessPendingErrorsTask.Wait();
+            Crashes.NotifyUserConfirmation(UserConfirmation.DontSend);
+
+            // Verify the logs have not been processed and have been removed.
+            _mockChannel.Verify(channel => channel.EnqueueAsync(It.Is<ManagedErrorLog>(log => log.Id == expectedManagedErrorLog.Id && log.ProcessId == expectedProcessId)), Times.Never());
```

Comment:
```suggestion
            _mockChannel.Verify(channel => channel.EnqueueAsync(It.Is<ManagedErrorLog>()), Times.Never());
```

Reply:
@achocron I committed this as-is and it broke. What do you want from this change?